interfaces/repo.go

```diff
@@ -736,9 +747,11 @@ func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string) []*Slo
 	if plug == nil {
 		return nil
 	}
-	if r.ifaces[plug.Interface].AutoConnect() == false {
+
+	if (r.ifaces[plug.Interface].AutoConnect() == false) && !(isSpecialAutoConnectSnap(plug.Snap)) {
 		return nil
 	}
+
 	var candidates []*Slot
 	for _, slotsForSnap := range r.slots {
 		for _, slot := range slotsForSnap {
```

Comment:
The logic below is also more special casing. This is getting a bit wild, and that's likely a side effect of a misdesign in our interface API. I believe we can easily sort all of these special cases out by having this instead:
```
iface := r.ifaces[plug.Interface]
...
        if iface.AutoConnect(plug, slot) {
                candidates = append(candidates, slot)
        }
```
We then move the custom logic to inside the respective interfaces.